### ISSUE TITLE
xcoverage_result.txt file empty unless -n option used

### ISSUE THREAD

**Issue by xross** [xmos/lib_xud#289]
For example:

$ pytest --smoke ./test_bulk_rx_basic.py --xcov --enabletracing 

Results in empty coverage file

$ pytest --smoke ./test_bulk_rx_basic.py --xcov --enabletracing -n=2

Generates coverage as expected

**Comment by kevinyeungxmos** [comment]
yeah, I also feel really handful cause the worker_id doesn't able to distinguish gw and master. So it should use -n even if running on single procces.

**Comment by kevinyeungxmos** [comment]
Fixed., the worker_id works now. see #292
